docs/UserGuide.md

```diff
@@ -72,43 +72,73 @@ Shows a message explaning how to access the help page.
-* `add n/Betsy Crowe t/friend e/betsycrowe@example.com a/Newgate Prison p/1234567 t/criminal`
+* `add n/John Wang p/98765432 e/johnwang@example.com a/John street, block 123, #01-01 g/M`
+* `add n/John p/92781123 e/john@example.com a/Donald street, block 248, #02-04 g/m`
+* `add n/Charlotte p/81286623 e/charlotte@example.com a/Charity street, block 101, #10-82  g/F`
 
 ### Listing all persons : `list`
 
 Shows a list of all persons in the address book.
 
 Format: `list`
 
-### Editing a person : `edit`
+### Editing a contact : `edit`
 
-Edits an existing person in the address book.
+Edits an existing contact in the address book.
 
-Format: `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [t/TAG]…​`
+Format: `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [g/GENDER] [d/DOB] [ph/PURCHASE_INDEX,UPDATED_PURCHASE_DETAIL] [t/TAG]`
 
-* Edits the person at the specified `INDEX`. The index refers to the index number shown in the displayed person list. The index **must be a positive integer** 1, 2, 3, …​
-* At least one of the optional fields must be provided.
-* Existing values will be updated to the input values.
-* When editing tags, the existing tags of the person will be removed i.e adding of tags is not cumulative.
-* You can remove all the person’s tags by typing `t/` without
-    specifying any tags after it.
+- Edits the person at the specified `INDEX`. The index refers to the index number shown in the displayed person list.
+  The index must be **a positive integer** 1, 2, 3, …​, and it must be within the range of the contact list index.
+- At least one of the optional fields must be provided.
+- Existing values will be updated to the input values.
+- When editing tags, the existing tags of the person will be removed (i.e adding of tags is not cumulative).
+- You can remove all the person’s tags by typing t/ without specifying any tags after it.
 
 Examples:
-*  `edit 1 p/91234567 e/johndoe@example.com` Edits the phone number and email address of the 1st person to be `91234567` and `johndoe@example.com` respectively.
+*  `edit 1 p/91234567 e/johndoe@example.com` Edits the phone number and email address of the 1st person to be
+   `91234567` and `johndoe@example.com` respectively.
 *  `edit 2 n/Betsy Crower t/` Edits the name of the 2nd person to be `Betsy Crower` and clears all existing tags.
+*  `edit 3 n/Charlotte g/F d/3.4.1998 ph/3,[20.03.2022, 101.25, Chair, Table, Banana]` Edits the 3rd person’s contact:
+   edits name to be `Charlotte`, edits gender to be `Female`, edits date of birth to be `3.4.1998`,
+   edits the purchase history to be `3,[20.03.2022, 101.25, Chair, Table, Banana]`.
+
+__Optional Parameter 1: Gender__
+
+Format: `edit INDEX [g/GENDER]`
+
+- Edits the gender of a person in the contact list. The genders accepted by the contact list are: M/m for male,  F/f for female.
+- INDEX must be positive integers (i.e 1,2,3…)
+- INDEX must be within the range of the contact list index (i.e from 1 to size of contact list).
```

Comment:
Perhaps format INDEX as inline code as well?

Reply:
Okk I will change this, thanks for the review!